dspace-api/src/main/java/org/dspace/checker/dao/impl/MostRecentChecksumDAOImpl.java

```diff
@@ -43,58 +45,54 @@ protected MostRecentChecksumDAOImpl() {
     @Override
     public List<MostRecentChecksum> findByNotProcessedInDateRange(Context context, Date startDate, Date endDate)
         throws SQLException {
-//                    + "most_recent_checksum.last_process_start_date, most_recent_checksum.last_process_end_date, "
-//                    + "most_recent_checksum.expected_checksum, most_recent_checksum.current_checksum, "
-//                    + "result_description "
-//                    + "from checksum_results, most_recent_checksum "
-//                    + "where most_recent_checksum.to_be_processed = false "
-//                    + "and most_recent_checksum.result = checksum_results.result_code "
-//                    + "and most_recent_checksum.last_process_start_date >= ? "
-//                    + "and most_recent_checksum.last_process_start_date < ? "
-//                    + "order by most_recent_checksum.bitstream_id
-
-        Criteria criteria = createCriteria(context, MostRecentChecksum.class);
-        criteria.add(
-            Restrictions.and(
-                Restrictions.eq("toBeProcessed", false),
-                Restrictions.le("processStartDate", startDate),
-                Restrictions.gt("processStartDate", endDate)
-            )
+
+        CriteriaBuilder criteriaBuilder = getCriteriaBuilder(context);
+        CriteriaQuery<MostRecentChecksum> criteriaQuery = getCriteriaQuery(criteriaBuilder, MostRecentChecksum.class);
+        Root<MostRecentChecksum> mostRecentChecksumRoot = criteriaQuery.from(MostRecentChecksum.class);
+        criteriaQuery.select(mostRecentChecksumRoot);
+        criteriaQuery.where(criteriaBuilder.and(
+            criteriaBuilder.equal(mostRecentChecksumRoot.get(MostRecentChecksum_.toBeProcessed), false),
+            criteriaBuilder
+                .lessThanOrEqualTo(mostRecentChecksumRoot.get(MostRecentChecksum_.processStartDate), startDate),
+            criteriaBuilder.greaterThan(mostRecentChecksumRoot.get(MostRecentChecksum_.processStartDate), endDate)
+                            )
         );
-        criteria.addOrder(Order.asc("bitstream.id"));
-        return list(criteria);
+        List<Order> orderList = new LinkedList<>();
+        orderList.add(criteriaBuilder.asc(mostRecentChecksumRoot.get(MostRecentChecksum_.bitstream)));
```

Comment:
Does this work?  I haven't found anything to say that `MostRecentChecksum_.bitstream` means `Bitstream.id`.

Reply:
The MostRecentChecksum_ class will make static fields that are named the same as they are in the model class. So the `MostRecentChecksum_.bitstream` in this instance refers to the `private Bitstream bitstream` field in MostRecentChecksum.java which in turn refers to the bitstream_id column. If my interpretation/deduction is correct, this should mean that the bitstream id is used.